### ISSUE TITLE
LocalStorage use for Auth system

### ISSUE THREAD

**Issue by cfjedimaster** [driftyco/ionic-cloud-angular#30]
The default behavior for logging in via Auth is to store "email" and "password" in LocalStorage. Ionic should really go out of their way to avoid using keys that an end user may use in their own application. As it stands, this is really two issues:

a) You document that you persist logins, but you do not *how* - and that is bad.
b) WHen you do make use of LocalStorage, you should be consistent. I see a key called "ionic_insights_session". But my user details are in "email" and "password". Why not "ionic_email" and "ionic_password"?

**Comment by imhoffd** [comment]
We are definitely not storing passwords in LocalStorage. Passwords shouldn't be stored anywhere except in their owner's brains.

We do store a JWT token in LocalStorage after authenticating the user, [as seen here](https://github.com/driftyco/ionic-cloud/blob/2242d3d4a743bd9ddd71736ed45c556d59f69776/src/auth.ts#L249-L257). We store the token in a namespaced key with the app ID.

These are the keys I see in my app:

![image](https://cloud.githubusercontent.com/assets/236501/20040758/d1831ec4-a422-11e6-9157-7fa27c321bf4.png)

As for documenting how auth works, I totally agree! I've added a task in our issue tracker to document more how our auth system actually works.

**Comment by cfjedimaster** [comment]
You are absolutely right. I forgot I had stored it myself and then removed the code that did it. While I still think it needs documenting (and y'all will so that's good!) I apologize for the misfire here!

**Comment by imhoffd** [comment]
@cfjedimaster Speaking of LocalStorage for auth, do you see any issues with continuing to use LocalStorage (as opposed to https://github.com/driftyco/ionic-storage)? Just curious on your thoughts.

**Comment by cfjedimaster** [comment]
I think it depends. With iOS killing off LS randomly, Ionic Storage may be a bit safer.

My #1 issue though is documenting and having a clear "line in the sand" for Ionic junk. That way a developer never accidentally intrudes on your stuff.
